loki/transform/transform_inline.py

```diff
@@ -193,7 +285,7 @@ def inline_elemental_functions(routine):
     # Remove all module imports that have become obsolete now
     import_map = {}
     for im in FindNodes(Import).visit(routine.spec):
-        if all(hasattr(s, 'type') and s.type.dtype in removed_functions for s in im.symbols):
+        if im.symbols and all(hasattr(s, 'type') and s.type.dtype in removed_functions for s in im.symbols):
```

Comment:
I think the `hasattr(s, 'type')` should no longer be necessary - that is a remnant of the times when we represented the symbol list in `Import` with names as strings.